### DISCUSSION TITLE
could you explain the account interaction?

### DISCUSSION THREAD

**Discussion by ChaosNicro** [KRTirtho/spotube#72] [Q&A] [answered]
Hey there,
The project looks interesting, though I would like some more info on the login method.

Why is a dev-app connection used?
What access is granted to/used by the app, regarding the used account?
As what does spotify see/process the app?
What data is sent back to spotify using the connected account?

Thanks for making this.

**Comment by KRTirtho** [comment, marked as answer]
Of course, here's the answers:

1. Why is a dev-app connection used?
    => Spotify API doesn't allow to use an API key publicly. You've to manually add users that you want to allow to use your service & that number is also limited to 5-10 user in development. But it does allow using one's own account without any issues (that makes sense as its his/her own application)
2. What access is granted to/used by the app, regarding the used account?
    => Following scopes are used:
    https://github.com/KRTirtho/spotube/blob/5307cd77588dd0fee8e7cd196a19fd847ba2cc08/lib/components/Home/Home.dart#L33-L43
3. As what does spotify see/process the app?
    => An application in development
4. What data is sent back to spotify using the connected account?
    => Literally, nothing. Even the playback data stays on user's machine. Spotify won't even know what you are/were listening
